app/diagnostics.go

```diff
@@ -0,0 +1,394 @@
+	TRACK_LICENSE  = "license"
+	TRACK_SERVER   = "server"
+)
+
+var client *analytics.Client
+
+func SendDailyDiagnostics() {
+	if *utils.Cfg.LogSettings.EnableDiagnostics {
+		initDiagnostics()
+		trackActivity()
+		trackConfig()
+		trackLicense()
+		trackServer()
+	}
+}
+
+func initDiagnostics() {
+	if client == nil {
+		client = analytics.New(SEGMENT_KEY)
+		client.Identify(&analytics.Identify{
+			UserId: utils.CfgDiagnosticId,
+		})
+	}
+}
+
+func SendDiagnostic(event string, properties map[string]interface{}) {
+	client.Track(&analytics.Track{
+		Event:      event,
+		UserId:     utils.CfgDiagnosticId,
+		Properties: properties,
+	})
+}
+
+func isDefault(setting interface{}, defaultValue interface{}) bool {
+	if setting == defaultValue {
+		return true
+	}
+	return false
+}
+
+func trackActivity() {
+	var userCount int64
+	var activeUserCount int64
+	var inactiveUserCount int64
+	var teamCount int64
+	var publicChannelCount int64
+	var privateChannelCount int64
+	var directChannelCount int64
+	var deletedPublicChannelCount int64
+	var deletedPrivateChannelCount int64
+	var postsCount int64
+
+	if ucr := <-Srv.Store.User().GetTotalUsersCount(); ucr.Err == nil {
+		userCount = ucr.Data.(int64)
+	}
+
+	if ucr := <-Srv.Store.Status().GetTotalActiveUsersCount(); ucr.Err == nil {
+		activeUserCount = ucr.Data.(int64)
+	}
+
+	if iucr := <-Srv.Store.Status().GetTotalActiveUsersCount(); iucr.Err == nil {
+		inactiveUserCount = iucr.Data.(int64)
+	}
+
+	if tcr := <-Srv.Store.Team().AnalyticsTeamCount(); tcr.Err == nil {
+		teamCount = tcr.Data.(int64)
+	}
+
+	if ucc := <-Srv.Store.Channel().AnalyticsTypeCount("", "O"); ucc.Err == nil {
+		publicChannelCount = ucc.Data.(int64)
+	}
+
+	if pcc := <-Srv.Store.Channel().AnalyticsTypeCount("", "P"); pcc.Err == nil {
+		privateChannelCount = pcc.Data.(int64)
+	}
+
+	if dcc := <-Srv.Store.Channel().AnalyticsTypeCount("", "D"); dcc.Err == nil {
+		directChannelCount = dcc.Data.(int64)
+	}
+
+	if duccr := <-Srv.Store.Channel().AnalyticsDeletedTypeCount("", "O"); duccr.Err == nil {
+		deletedPublicChannelCount = duccr.Data.(int64)
+	}
+
+	if dpccr := <-Srv.Store.Channel().AnalyticsDeletedTypeCount("", "P"); dpccr.Err == nil {
+		deletedPrivateChannelCount = dpccr.Data.(int64)
+	}
+
+	if pcr := <-Srv.Store.Post().AnalyticsPostCount("", false, false); pcr.Err == nil {
+		postsCount = pcr.Data.(int64)
+	}
+
+	SendDiagnostic(TRACK_ACTIVITY, map[string]interface{}{
+		"registered_users":          userCount,
+		"active_users":              activeUserCount,
+		"registered_inactive_users": inactiveUserCount,
+		"teams":                     teamCount,
+		"public_channels":           publicChannelCount,
+		"private_channels":          privateChannelCount,
+		"direct_message_channels":   directChannelCount,
+		"public_channels_deleted":   deletedPublicChannelCount,
+		"private_channels_deleted":  deletedPrivateChannelCount,
+		"posts":                     postsCount,
+	})
+}
+
+func trackConfig() {
```

Comment:
That's deliberate. We want to only include config settings once the correct level of detail for each new setting has been considered in terms of the privacy implications. We don't want to inadvertently start including the full text of a new config variable where it might only be appropriate to include a boolean for "is default" of it.